californium-core/src/main/java/org/eclipse/californium/core/coap/Option.java

```diff
@@ -44,8 +44,24 @@
  * UnSafe = (onum &amp; 2);
  * NoCacheKey = ((onum &amp; 0x1e) == 0x1c);
  * </pre></blockquote><hr>
- * 
- * CoAP defines several option numbers {@link OptionNumberRegistry}.
+ *
+ * {@code CoAP} defines several option numbers detailed in {@link OptionNumberRegistry}.
+ * <p>
+ * Class variables {@code number} and {@code value} directly maps to {@code CoAP} request
+ * and response options as is. In {@code CoAP} specification {@code number} is
+ * an option header key as {@code int} and {@code value} is represented as
+ * a raw byte array {@code byte[]}. User must be careful when using {@code value} directly
+ * as depending on actual option, {@code value} may be {@code empty}, {@code opaque},
+ * {@code uint} or {@code string}. For example, for {@code uint} the number 0 is represented
+ * with an empty option value (a zero-length sequence of bytes) and the number 1 by a single
+ * byte with the numerical value of 1 (bit combination 00000001 in most significant bit first
+ * notation). A recipient MUST be prepared to process values with leading zero bytes.
+ * <p>
+ * {@code Option} has helper methods, namely {@link #getIntegerValue()} and
+ * {@link #toValueString()} taking into account actual option type and how it
```

Comment:
`toValueString()` is more a textual description (maybe sued for logging). For string options, `getStringValue()` is the right one to access the value with the proper type. So I would prefer to have `getStringValue()` mentioned here instead of the `toValueString()`.